Sources/BraintreePayPalNativeCheckout/BTPayPalNativeCheckoutRequest.swift

```diff
@@ -2,12 +2,51 @@
 import BraintreePayPal
 #endif
 
-/// Options for the PayPal Checkout and PayPal Checkout with Vault flows.
-@objc public class BTPayPalNativeCheckoutRequest: BTPayPalCheckoutRequest, BTPayPalNativeRequest {
+///  Payment intent.
+///
+///  - Note: Must be set to `.sale` for immediate payment, `.authorize` to authorize a payment for capture later, or `.order` to create an order. Defaults to `authorize`.
+///  - SeeAlso: see https://developer.paypal.com/docs/integration/direct/payments/capture-payment/ Capture payments later
+///  - SeeAlso: https://developer.paypal.com/docs/integration/direct/payments/create-process-order/ Create and process orders
+@objc public enum BTPayPalNativeRequestIntent: Int {
+    /// Authorize
+    case authorize
 
-    let paymentType: BTPayPalPaymentType = .checkout
+    /// Sale
+    case sale
 
-    let hermesPath: String = "v1/paypal_hermes/create_payment_resource"
+    /// Order
+    case order
+}
+
+/// Options for the PayPal Checkout flow.
+@objcMembers public class BTPayPalNativeCheckoutRequest: BTPayPalNativeRequest {
+    
+    // MARK: - Public Properties
+    // next_major_version: subclass BTPayPalCheckoutRequest once BraintreePayPal is in Swift.
+
+    /// Optional: Payment intent. Defaults to `.authorize`. Only applies to PayPal Checkout.
+    public var intent: BTPayPalNativeRequestIntent
+    
+    /// Used for a one-time payment.
+    ///
+    /// Amount must be greater than or equal to zero, may optionally contain exactly 2 decimal places separated by '.' and is limited to 7 digits before the decimal point.
+    public let amount: String
+    
+    /// Optional: Offers PayPal Pay Later if the customer qualifies. Defaults to false. Only available with PayPal Checkout.
+    public var offerPayLater: Bool
```

Comment:
The bool is defaulted in the init. Is this our preferred way to default values? We cannot use an optional `Bool` here since those cannot be represented in Objective-C.

Reply:
I think this works well!

Once we're all in Swift, we should do a default `nil` audit since we'll no longer be constrained by our existing ObjC behavior.